src/NumericalAlgorithms/Spectral/Mesh.hpp

```diff
@@ -250,9 +229,25 @@ class Mesh {
   void pup(PUP::er& p);
 
  private:
-  Index<Dim> extents_{};
-  std::array<Spectral::Basis, Dim> bases_{};
-  std::array<Spectral::Quadrature, Dim> quadratures_{};
+  static Spectral::Basis extract_basis(uint8_t bits);
+  static Spectral::Quadrature extract_quadrature(uint8_t bits);
+
+  template <size_t LocalDim>
+  friend class Mesh;
+
+  template <size_t LocalDim>
+  friend bool operator==(const Mesh<LocalDim>& lhs, const Mesh<LocalDim>& rhs);
+
+  // - 8 bits per extent = 3 * 8 = 24 bits = 3 bytes
+  //   This limits us to 128 extent per direction. Since FD is 2x DG, this
+  //   means at most 64 DG points, which is should be fine.
+  // - We encode the basis & quadrature as two sets of 4 bits in a uint8_t.
+  std::array<uint8_t, 6> bit_field_{
```

Comment:
I think this would be easier to read if you had two arrays of size 3 instead of one array of size 6 with the first and second halves interpreted differently.